File: 백준/Silver/1929. 소수 구하기/소수 구하기.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int M = sc.nextInt(), N = sc.nextInt();

        sc.close();

        boolean[] prime = new boolean[N + 1];
        prime[0] = prime[1] = true; // 0과 1은 소수에서 제외

        for (int i = 2; i * i <= N; i++) {
            if (!prime[i]) { // prime[i]가 소수라면
                for (int j = i * i; j <= N; j += i) { // 소수가 아닌 것을 true로 표시
                    prime[j] = true;
                }
            }
        }

        StringBuilder sb = new StringBuilder();

        for(int i = M; i <= N; i++){
            if(!prime[i]){
                sb.append(i).append("\n");
            }
        }

        System.out.println(sb);
    }
}

File: 백준/Silver/1929. 소수 구하기/소수 구하기.kt
import java.util.*
import java.io.*
import kotlin.math.sqrt

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {

    val st = StringTokenizer(readLine())
    val M = st.nextToken().toInt()
    val N = st.nextToken().toInt()
    close()

    val isPrime = BooleanArray(N + 1){true}
    isPrime[0] = false
    isPrime[1] = false

    // 에라토스테네스의 체 (2부터 N의 제곱근 값까지 iteration)
    for(i in 2..sqrt((N + 1).toDouble()).toInt()) {
        // isPrime[i] 값이 소수면 소수의 배수는 소수가 되지 않기 때문에 모두 false 처리
        if(isPrime[i]) {
            for(j in i*i..N step i) {
                isPrime[j] = false
            }
        }
    }

    BufferedWriter(OutputStreamWriter(System.out)).run {
        for(i in M .. N) {
            if(isPrime[i]) write("$i\n")
        }
        flush()
        close()
    }
}